### PR TITLE
feat(Protractor): add e2e testing via protractor

### PR DESCRIPTION
**Pull request by NathanWalker**

### CODE REVIEW

File: app/components/app/app.e2e.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+describe('App', function() {
```
Comment (mgechev) [inline app/components/app/app.e2e.js]: Can we rename it to `app.e2e.ts` in order to have autocompletion? I think we already have protractor type definitions.

File: protractor.conf.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+exports.config = {
+    baseUrl: 'http://localhost:5555/dist/dev',
+
+    specs: [
+        'app/**/*.e2e.js'
```
Comment (mgechev) [inline protractor.conf.js]: If we change the e2e file to `ts` this should become `dist/dev/**/*.e2e.js`.

**Comment by mgechev** [comment]
Great job! I added two small things.

**Comment by NathanWalker** [comment]
Great suggestions @mgechev thanks! I also added more e2e tests as well to complete the coverage; they should serve as good examples of how to text various behaviors :+1:

**Comment by mgechev** [comment]
@NathanWalker awesome! I'll merge tomorrow morning.

**Comment by ludohenin** [comment]
Awesome

**Pull request merged**